### DISCUSSION TITLE
TLS handshake failed; No server keys configured

### DISCUSSION THREAD

**Discussion by enboig** [angristan/openvpn-install#1099] [Help]
When I try to connect it fails.
Router and NAT should be ok because I see connection try in server log.
I checked configuration using webmin and it tells me "No server keys configured".

Could this be a "password" problem when installing openvpn? I set a password for key and cert.
```
port 1194
proto udp
dev tun
user nobody
group nogroup
persist-key
persist-tun
keepalive 10 120
topology subnet
server 10.8.0.0 255.255.255.0
ifconfig-pool-persist ipp.txt
push "dhcp-option DNS 94.140.14.14"
push "dhcp-option DNS 94.140.15.15"
push "redirect-gateway def1 bypass-dhcp"
dh none
ecdh-curve prime256v1
tls-crypt tls-crypt.key
crl-verify crl.pem
ca ca.crt
cert server_M8wGr7nZ6zD3KNa1.crt
key server_M8wGr7nZ6zD3KNa1.key
auth SHA256
cipher AES-128-GCM
ncp-ciphers AES-128-GCM
tls-server
tls-version-min 1.2
tls-cipher TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256
client-config-dir /etc/openvpn/ccd
status /var/log/openvpn/status.log
verb 3
```

**Comment by TinCanTech** [comment]
> Could this be a "password" problem when installing openvpn? I set a password for key and cert.

Then your config needs command `askpass`, so that you can enter the password for the key or store the password in a file.
